### ISSUE TITLE
Catboost cv function models are different from trained via fit in loop

### ISSUE THREAD

**Issue by FedorTarasow** [catboost/catboost#2695]
Problem: Catboost cv function models are different from trained via fit in loop
catboost version: 1.2.5
Operating System: Kaggle
CPU: 30GB
GPU:
Hi I am trying to compare results of cross-falidation catboost.cv models with just simple models fit with loop.

Way to train models via loop:
`model=CatBoostClassifier(iterations=100,loss_function='Logloss',eval_metric='AUC',random_state=888, logging_level ='Silent',learning_rate=0.029999999329447743)
models=[]
for i in range(0,5):
    train_pool=Pool(data=train_df.iloc[folds[i][0]][features],label=train_df.iloc[folds[i][0]]['Credit_Score'],group_id=train_df.iloc[folds[i][0]]['Customer_ID'],cat_features=cat_feats)
    val_pool=Pool(data=train_df.iloc[folds[i][1]][features],label=train_df.iloc[folds[i][1]]['Credit_Score'],group_id=train_df.iloc[folds[i][1]]['Customer_ID'],cat_features=cat_feats)
    model.fit(train_pool)
    models.append(model.copy())`

Way to train models with catboost vb:
`cv_data=cv(params={'iterations':100,'loss_function':'Logloss','eval_metric':'AUC','random_state':888,'thread_count':1},
          pool=train_pool,
          return_models=True,logging_level='Silent',folds=iter(folds))`

Gini based on the predictions of models are different.

Please have a look at [kaggle notebook ](https://www.kaggle.com/code/fedort90/cv-question) that demostrates the issue.
P.S. Learning rate from model in loop fitting was taken from cv model.get_all_params()

**Comment by ek-ak** [comment]
Hello!
There are several other parameters besides `learning_rate` that are different. If you set parameters like these
```
params = cv_data[-1][0].get_all_params()
for key in ('force_unit_auto_pair_weights', 'bayesian_matrix_reg', 'pool_metainfo_options', 'permutation_count'):
    if key in params:
        del params[key]
model = CatBoostClassifier(**params)
```
the results will be very similar.

**Comment by FedorTarasow** [comment]
Hi!
I have used **params in my code and models still show different results. (I updated Kaggle noteboook)
